kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java

```diff
@@ -559,6 +572,21 @@ private void push(UserNotification un) {
         });
     }
 
+    /**
+     * Clear invalid push subscription for a user when push service returns permanent failure
+     */
+    private void clearUserPushSubscription(String userId) {
+        try {
+            List<PushSubscription> subscriptions = pushSubscriptionRepository.findByUser(userId);
+            for (PushSubscription subscription : subscriptions) {
+                pushSubscriptionRepository.delete(subscription);
+                log.debug("Removed invalid push subscription {} for user {}", subscription.getEndpoint(), userId);
+            }
+        } catch (Exception e) {
+            log.error("Failed to clear push subscription for user {}: {}", userId, e.toString());
```

Comment:
Did you see any exceptions here? It seems weird to get one at this point? If you don't need the try you can just make this a lambda, or, better still, add a batch delete to the repository. That can happen later though.

Reply:
no i didn't run into any exceptions. i will remove